pkg/scheduler/core/generic_scheduler.go

```diff
@@ -734,7 +734,10 @@ func PrioritizeNodes(
 			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
 		}
 	}
-
+	
+	// Wait for results array initialization
+	wg.Wait()
```

Comment:
At L728 and L749, the index for `results[]` is different, so there is no data race, and do not need to add it.